tools/plugins/network/src/main/kotlin/net/corda/cli/plugins/network/OnBoardMember.kt

```diff
@@ -195,79 +208,55 @@ class OnBoardMember : Runnable, BaseOnboard() {
 
     override val registrationContext by lazy {
         val preAuth = preAuthToken?.let { mapOf("corda.auth.token" to it) } ?: emptyMap()
-        
-        if (customProperties.size > 100) {
-            throw IllegalArgumentException("Cannot specify more than 100 key-value pairs")
-        }
-        
-        val notaryServiceName = customProperties["corda.notary.service.name"]
-      
-        customProperties = customProperties.filterKeys { it != "corda.notary.service.name" }
-        
+
         val extProperties = customProperties.filter {
             val key = it.key
             val value = it.value
-            val keyAndValueNotEmpty = (key.isNotEmpty()).and(value.isNotEmpty())
-            val keyStartsWithExt = key.startsWith("ext.")
-            val keyDoesNotEndWithExt = !key.endsWith("ext.")
-            val allChecks = keyAndValueNotEmpty.and(keyStartsWithExt).and(keyDoesNotEndWithExt)
+            val keyStartsWithExt = key.startsWith("$CUSTOM_KEY_PREFIX.")
```

Comment:
Curious what others think about these checks here. But I feel if we do want to keep the key value length checks we should also keep the check on number of fields. And in that case, applying them like this would probably be more readable -
```suggestion
val extProperties = customProperties.filterKeys { it.startsWith("$CUSTOM_KEY_PREFIX.") }.apply {
            if (size > 100) {
                throw IllegalArgumentException("The number of custom fields ($size) provided is larger than the maximum allowed ($MAX_CUSTOM_FIELDS).")
            }
            val keysTooLong = filterKeys { it.length > MAX_KEY_LENGTH }
            if (keysTooLong.isNotEmpty()) {
                throw IllegalArgumentException("The following keys exceed the maximum allowed length ($MAX_KEY_LENGTH): ${keysTooLong.keys}")
            }
            val valuesTooLong = filterValues { it.length > MAX_VALUE_LENGTH }
            if (valuesTooLong.isNotEmpty()) {
                throw IllegalArgumentException("Values of the following keys exceed the maximum allowed length ($MAX_VALUE_LENGTH): ${keysTooLong.keys}")
            }
        }
```

Reply:
No strong opinion here, but if these are enforced by the platform already I don't think it's a big issue not enforcing them at the plugin level as well. They will lead to an invalid/declined registration anyway, which might require the user a few more steps to identify what went wrong. 

But, on the other hand they lead to quite a bit of duplication and potential of drift against the rules in the platform that can change. This can be eliminated by having shared code between the plugin & the platform, but I think at this point we are probably starting to over-engineer things.